### ISSUE TITLE
License ? in particular unet

### ISSUE THREAD

**Issue by VolkerH** [CellProfiler/CellProfiler-plugins#72]
Hi Tim @cells2numbers , 
what is the license for the plugin code?

In particular I'm interested in the license for the `unet` code and the pre-trained model (the model is downloaded from a google drive which is somewhat obscure).

The reason I'm asking is that over the last couple of days I took the code, added a few improvements (changed deprecated Keras syntax, included some of the suggestions by @eric-czech, added automated rescaling of input images to model size and rescaling of classification results to input size and turned it into a pip-installable module that includes the pre-trained model.

If the license allows, I would like to put the code on GitHub. I also would have to know who should be credited for creating the module and who should be cited.

**Comment by cells2numbers** [comment]
Hey Volker, 

that sounds great! Could you send a link to the project / webpage? I'd be happy to test it :) 

Please note that the original UNet code comes from https://github.com/carpenterlab/unet4nuclei which was used in https://www.biorxiv.org/content/early/2018/05/31/335216 and written by jccaicedo  (https://github.com/jccaicedo). So please cite this publication! 

Regarding a license: I just added a BSD3 license to the classifypixelunet modul ( it is the same license that we use for CellProfiler). The license states how you should handle credits etc. 

The weights are under the same license but have not been published somewhere other than in the google drive. I'll open another ticket for that! 

T

**Comment by VolkerH** [comment]
@cells2numbers 
I finally got around to putting up this package: https://github.com/VolkerH/unet-nuclei
Now that I figured out how to run the network on my laptops GPU using plaidML I'll be using it more often.
Comments are welcome.

**Comment by cells2numbers** [comment]
@VolkerH  Great, awesome to have an installable python package! I'll try it later. Can the weights be easily changed? That would be awesome! 

I will look into the code again once I find some time. I think we could replace the rescaling with some image padding to get the correct image size because of the rescaling sometimes smoothes the image data. I will change the code for this plugin and try to add it your package!

**Comment by VolkerH** [comment]
@cells2numbers
You are right about the padding vs. scaling. Scaling does indeed blur the
nuclei boundaries. However, at the time I started bundling this I was
running the network on my laptop's CPU, which was effectively limiting me
to very small image sizes. So being able to initialize a smaller network
and then have the images downscaled automatically to fit the network was
quite welcome.

Now that I figured out how to run this on the GPU of my laptop with PlaidML
larger images no longer pose a problem.

Note that the code requires python 3.6. I should add this to the
requirements.

Yes, you can use different weights by providing the optional parameter
`weights_filename`.

```
def unet_initialize(input_shape, weights_filename = _model_path,
automated_shape_adjustment=False):
```

I should probably add a bit more documentation.
If you have other pre-trained models that could be bundled with it, let me
know.
